src/components/Articles/EditArticle.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
+import Loader from 'react-loader';
+import {
+  fetchSpecificArticle,
+  updateArticle,
+} from '../../actions/articleActions';
+import Article from './Article';
+import CreateArticleForm from './CreateArticleForm';
+
+export class EditArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      body: '',
+      isEditing: false,
+    };
+  }
+
+  componentDidMount() {
+    const { fetchSpecificArticle } = this.props;
+    const {
+      match: {
+        params: { slug },
+      },
+    } = this.props;
+    fetchSpecificArticle(slug);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { articlePayload } = nextProps;
+    if (nextProps.articlePayload !== this.props.articlePayload) {
+      const { article: { title, body, description } } = articlePayload;
+      this.setState({ title, body, description });
+    }
+    if (nextProps.editArticleSuccess === true) {
+      this.setState({ isEditing: false });
+      const { history } = this.props;
+      history.push('/dashboard');
+    }
+  }
+
+  handleChange = event => {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleEditorChange = value => {
+    this.setState({ body: value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const { title, description, body } = this.state;
+    const payload = {
+      article: {
+        title,
+        description,
+        body,
+      },
+    };
+
+    const { updateArticle } = this.props;
+    const {
+      match: {
+        params: { slug },
+      },
+    } = this.props;
+    updateArticle(slug, payload);
+  };
+
+  resetForm = () => {
+    this.setState({
+      title: '',
+      description: '',
+      body: '',
+    });
+  };
+
+  toggleEdit = () => {
+    const { isEditing } = this.state;
+    this.setState({ isEditing: !isEditing });
+  }
+
+  render() {
+    const {
+      title,
+      description,
+      body,
+      isEditing,
+    } = this.state;
+    const { loading, articlePayload, articlePayload: { article } } = this.props;
+    if (isEditing) {
+      return (
+        <CreateArticleForm
```

Comment:
Rename this to `ArticleForm`. `Create` does not sound right in `Edit`.

Reply:
`CreateArticleForm` has been renamed to `ArticleForm`